docs/src/capi/explanation/in-place-upgrades.md

```diff
@@ -0,0 +1,236 @@
+they’re created. In order to change a machine, according to CAPI design 
+decisions, we must replace it with a new one. Quoting from [CAPI concepts][1]:
+
+> From the perspective of Cluster API, all Machines are immutable: once 
+they are created, they are never updated (except for labels, annotations and 
+status), only deleted.
+
+{{product}} CAPI leverages the fact that "annotations" can be changed on 
+an already created machine. As we will later see in details, we can perform 
+an in-place upgrade by changing a machine’s annotations. Changing “labels” 
+or “status” does not make much sense, as the former is mostly used as a 
+[grouping/organization mechanism][2] and the latter is [supplied and updated 
+by Kubernetes][3].
+
+That being said, the whole idea of doing an in-place upgrade might not be 
+completely aligned with the upstream cluster API design decisions as we’re 
+essentially changing a machine’s Kubernetes version, something that’s 
+supposed to be achieved by doing a rolling upgrade (replacing old ones). 
+Either way, having the ability to perform an in-place upgrade comes with 
+many benefits, so we decided to design and implement a way to enable users 
+perform these upgrades with minimum friction.
+
+## Controllers
+
+In {{product}} CAPI, we have two main types of controllers that handle the 
+process of performing in-place upgrades:
+
+- Single Machine In-Place Upgrade Controller
+- Orchestrated In-Place Upgrade Controller
+
+The core component of performing an in-place upgrade is the “Single Machine 
+Upgrader”. It watches for certain annotations on machines and reconciles 
+them to make sure the upgrades happen as expected. As its name suggests, 
+it reconciles each machine individually, considers the machines separately 
+and does not assume any sort of relation between them.
+
+The “Orchestrator” on the other hand, watches for certain annotations on 
+machine owners, reconciles them and upgrades groups of owned machines. 
+It’s responsible for making sure that all the machines owned by the 
+reconciled object get upgraded successfully.
+
+The main annotations that drive the upgrade process are as follows (for a 
+complete and up-to-date list of these annotations and their values please 
+refer to [annotations reference page][4]):
+
+- `v1beta2.k8sd.io/in-place-upgrade-to` : Instructs the controller to 
+perform an upgrade with the specified option/method. This is the only 
+annotation that we as the users need to put on the objects.
+- `v1beta2.k8sd.io/in-place-upgrade-status`: As soon as the controller 
+starts the upgrade process, the object will be marked with this 
+annotation to indicate the status of the upgrade. It can either be 
+`in-progress`, `failed` or `done`.
+- `v1beta2.k8sd.io/in-place-upgrade-release`: When the upgrade is 
+performed successfully, this annotation will indicate the current 
+Kubernetes release/version installed on the machine.
```

Comment:
Let's make this focused on what the user needs to know. This can probably be half in length once condensed.